python/cudf/cudf/core/series.py

```diff
@@ -2756,7 +2756,7 @@ def nunique(self, method="sort", dropna=True):
             raise NotImplementedError(msg)
         if self.null_count == len(self):
             return 0
-        return self._column.distinct_count(method, dropna)
+        return super().nunique(method, dropna)
```

Comment:
The implementation in `Series` probably isn't necessary at all. Since `Series` inherits from `SingleColumnFrame`, the implementation there should just work for `Series`. I would say to delete this function entirely. You can move the `if self.null_count == len(self)` check into the `SingleColumnFrame` implementation so that it still happens and will also be used for index types. We can safely get rid of the `method != "sort"` check because the same check is performed in the column implementation that gets called when you do `self._column.distinct_count(...)`, see [here](https://github.com/rapidsai/cudf/blob/branch-22.04/python/cudf/cudf/core/column/column.py#L815).